runtime/near-vm-runner/src/logic/logic.rs

```diff
@@ -2106,6 +2106,10 @@ impl<'a> VMLogic<'a> {
 
     /// Submits the data for a promise which is awaiting its value. See `promise_yield_create`.
     ///
+    /// Returns `1` if submitting the payload for the data dependency was successful. This
+    /// guarantees that the yield callback function will be executed with a payload. Otherwise a
```

Comment:
```suggestion
    /// guarantees that the yield callback function will be executed with given payload. Otherwise a
```

Reply:
NB: this "a" is pretty intentional, https://near.zulipchat.com/#narrow/stream/295558-pagoda.2Fcore/topic/Yield.2Fresume.20implementation/near/424061559, but let me know if you think we should still apply this suggestion.